### PR TITLE
7.0 mis builder backport from 8.0

### PR DESCRIPTION
**Pull request by lmignon**
Back port mis_builder from 8.0 (#86)

### CODE REVIEW

**Comment by gurneyalex** [comment]
My PR fixing the travis and runbot builds on the 8.0 branch were recently merged. Can you rebase your PR?

**Comment by lmignon** [comment]
@gurneyalex done.

**Comment by gurneyalex** [comment]
I let you backport the fixes asked on #86 when they get applied in there :smile_cat:

**Comment by JordiBForgeFlow** [comment]
I have applied the fixes asked on #86, in https://github.com/acsone/account-financial-reporting/pull/7. Please @lmignon review and merge if you agree.

**Comment by sebalix** [comment]
I implemented this module on a 7.0 instance for some statistics unrelated to accounting, it works like a charm (with SUM and AVG queries). :+1:

**Comment by sbidoul** [comment]
rebased

**Comment by pedrobaeza** [comment]
Can you squash the commit history. 179 commit seems a lot of commits.

**Comment by sbidoul** [comment]
@pedrobaeza the commits are in the 8.0 branch. And there are several authors.

Not sure why travis is complaining. Anyway errors seem unrelated so :+1: 

Thanks @jbeficent for the last effort.

**Comment by sbidoul** [comment]
> Can you squash the commit history. 179 commit seems a lot of commits.
> 
> It's a lot of work too :wink:

**Comment by pedrobaeza** [comment]
All commits from different authors come together, so not too much, but I'm not going to force you.

**Comment by JordiBForgeFlow** [comment]
@sbidoul Not sure why Travis fails. It's not related to the changes made.

File: mis_builder/models/mis_builder.py [hunk 1]
```diff
@@ -0,0 +1,913 @@
...
+                obj_ids = obj.search(cr, uid, domain, context=context)
+                data = obj.read(
+                    cr, uid, obj_ids, field_names, context=context)
+                s = AutoStruct(count=len(data))
+                if query.aggregate == 'min':
+                    agg = min
+                elif query.aggregate == 'max':
+                    agg = max
+                elif query.aggregate == 'avg':
+                    agg = lambda l: sum(l) / float(len(l))
```
Comment (sebalix) [inline mis_builder/models/mis_builder.py]: I encounter a problem here, a division by 0 could happen if there is no rows to compute.
I fixed it with the following line:

``` python
agg = lambda l: l and sum(l) / float(len(l)) or 0
```

**Comment by JordiBForgeFlow** [comment]
Travis says:
mis_builder/**openerp**.py:25: [C8102(manifest-required-key), ] Missing required key "images" in manifest file

But in OCA maintainer-tools template: https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py I don't see this parameter being a requirement.

**Comment by nmoturi** [comment]
We been using MIS Builder V7 successfully and would be glad if its gets pushed to OCA asap.

Thanks,
Naran

File: mis_builder/models/mis_builder.py (hunk 1 above)
Comment (sbidoul) [inline mis_builder/models/mis_builder.py]: @sebalix this was a backport issue. It should be fixed with @jbeficent's last commit: https://github.com/OCA/account-financial-reporting/commit/30744ab287a634a189d2f51b7ee88bed93475686

**Comment by sbidoul** [comment]
I rebased. No idea why travis fails. Seems unrelated. IMO, this is good for merging.

**Comment by lmignon** [comment]
There are 2 pytlint errors 

```
======== Testing test_pylint ========
************* Module account_chart_report.__openerp__
account_chart_report/__openerp__.py:28: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\ '. String constant might be missing an r prefix.
account_chart_report/__openerp__.py:28: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\ '. String constant might be missing an r prefix.
```

And travis need to fixed. The tests fails since the dependencies are no more checked out in the directory exepected by `travis_run_tests`. We must declare the dependencies in a oca_dependencies.txt file.

**Comment by lmignon** [comment]
Pep8 issue is fixed and travis runs but it seems that we have a regression. https://travis-ci.org/OCA/account-financial-reporting/jobs/94815699#L335

```
2015-12-04 07:35:47,188 22916 TEST openerp_test openerp.modules.module: FAIL: test_datetime_conversion (openerp.addons.mis_builder.tests.test_mis_builder.test_mis_builder)
2015-12-04 07:35:47,188 22916 TEST openerp_test openerp.modules.module: Traceback (most recent call last):
2015-12-04 07:35:47,188 22916 TEST openerp_test openerp.modules.module: `   File "/home/travis/build/OCA/account-financial-reporting/mis_builder/tests/test_mis_builder.py", line 40, in test_datetime_conversion
2015-12-04 07:35:47,188 22916 TEST openerp_test openerp.modules.module: `     'The converted date time convert must contains hour')
2015-12-04 07:35:47,188 22916 TEST openerp_test openerp.modules.module: ` AssertionError: '2014-07-04' != '2014-07-04 22:00:00'
2015-12-04 07:35:47,188 22916 TEST openerp_test openerp.modules.module: ` - 2014-07-04
2015-12-04 07:35:47,189 22916 TEST openerp_test openerp.modules.module: ` + 2014-07-04 22:00:00
2015-12-04 07:35:47,189 22916 TEST openerp_test openerp.modules.module: `  : The converted date time convert must contains hour
2015-12-04 07:35:47,189 22916 TEST openerp_test openerp.modules.module: `
2015-12-04 07:35:47,189 22916 TEST openerp_test openerp.modules.module: Ran 2 tests in 0.029s
2015-12-04 07:35:47,189 22916 TEST openerp_test openerp.modules.module: FAILED
2015-12-04 07:35:47,189 22916 TEST openerp_test openerp.modules.module:  (failures=1)
```

**Comment by lmignon** [comment]
@sbidoul Travis is green. It's ready to be merged

File: mis_builder/models/mis_builder.py (hunk 1 above)
Comment (sebalix) [inline mis_builder/models/mis_builder.py]: @sbidoul @jbeficent thanks

**Comment by JordiBForgeFlow** [comment]
Thanks @lmignon and @sbidoul! :+1:

**Comment by sbidoul** [comment]
I'm merging now. Many thanks to all who contributed to this backport.

**Pull request merged**